### DISCUSSION TITLE
[General] solidity compilation error: Member "..." not found or not visible after argument-dependent lookup in contract

### DISCUSSION THREAD

**Discussion by sanu2002** [zkSync-Community-Hub/zksync-developers#835] [General]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

```
// SPDX-License-Identifier: MIT
pragma solidity >0.2.0 <0.9.0;

import {Script} from "lib/forge-std/src/Script.sol";
import {CodeConstants} from "script/abstarct.s.sol";
import {MockVRFCoordinator} from "script/Mockcontract.s.sol";

contract HelperConfig is Script, CodeConstants {
    struct NetworkConfig {
        uint256 entranceFee;
        uint256 interval;
        address vrfCoordinator;
        bytes32 gasLane;
        uint32 callbackGasLimit;
        uint256 subscriptionId;
    }

    NetworkConfig public  activeNetworkConfig;

    constructor() {
        if (block.chainid == ETH_SEPOLIA_CHAIN_ID) {
            activeNetworkConfig = getSepoliaEthConfig();
        } else if (block.chainid == LOCAL_CHAIN_ID) {
            activeNetworkConfig = getLocalConfig();
        } else {
            revert("Unsupported network");
        }
    }


	function getActiveNetworkConfig() external view returns (NetworkConfig memory) {
        return activeNetworkConfig;
    }


  

    function getSepoliaEthConfig() public pure returns (NetworkConfig memory) {
        return NetworkConfig({
            entranceFee: 0.01 ether, // 1e16
            interval: 30, // 30 seconds
            vrfCoordinator: 0x9DdfaCa8183c41ad55329BdeeD9F6A8d53168B1B,
            gasLane: 0x787d74caea10b2b357790d5b5247c2f63d1d91572a9846f780606e4d953677ae,
            callbackGasLimit: 500000, // 500,000 gas
            subscriptionId: 0 // Replace with your subscription ID
        });
    }

    function getLocalConfig() public returns (NetworkConfig memory) {
        MockVRFCoordinator mockVrfCoordinator = new MockVRFCoordinator();

        return NetworkConfig({
            entranceFee: 0.01 ether,
            interval: 30, 
            vrfCoordinator: address(mockVrfCoordinator),
            gasLane: "",
            callbackGasLimit: 500000,
            subscriptionId: 0 // No subscription ID for local
        });
    }
}


```

```
// SPDX-License-Identifier: MIT
pragma solidity >0.2.0 <0.9.0;

import {Script} from "lib/forge-std/src/Script.sol";
import {console} from "lib/forge-std/src/console.sol";
import {Raffle} from "src/Raffle.sol";
import {HelperConfig} from "script/Networkconfig.s.sol";

contract DeployRaffle is Script {
    function run() external {
        // Deploy the HelperConfig to get the network configuration
        HelperConfig helperConfig = new HelperConfig();
        HelperConfig.NetworkConfig memory networkConfig = helperConfig.getActiveNetworkConfig();

        // Log the network configuration
        console.log("Raffle deployment config:");
        console.log("Entrance Fee:", networkConfig.entranceFee);
        console.log("Interval:", networkConfig.interval);
        console.log("VRF Coordinator:", networkConfig.vrfCoordinator);

        // Start broadcasting transactions
        vm.startBroadcast();

        // Deploy the Raffle contract
        Raffle raffle = new Raffle(
            networkConfig.entranceFee,
            networkConfig.interval,
            networkConfig.vrfCoordinator
        );

        vm.stopBroadcast();

        // Log the deployed contract address
        console.log("Raffle deployed at:", address(raffle));
    }
}


```

![image](https://github.com/user-attachments/assets/db23bbe7-c595-4bee-8760-de8331bd0e4f)

Why i am facing this error everything is fine in the code ??

**Comment by alhenaai** [comment]
It appears you are encountering an issue while deploying your smart contracts, but the specific error message or problem you're facing isn't clear from your description. To better assist you, could you please provide more details about the error message or issue you are encountering during the deployment process?

Additionally, if you are using specific tools or environments (like Foundry, Truffle, or Hardhat) for your deployment, please specify as it might help in diagnosing the problem more accurately.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by sanu2002** [reply]
Yes i am using foundry here .

```
root@Anon:~/SCL# forge script script/deploy.s.sol   --rpc-url $ETH_SEPOLIA_URL   --private-key $PRIAVTE_KEY   --broadcast
[⠊] Compiling...
[⠔] Compiling 30 files with Solc 0.8.27
[⠒] Solc 0.8.27 finished in 368.07ms
Error: Compiler run failed:
Error (9582): Member "getActiveNetworkConfig" not found or not visible after argument-dependent lookup in contract HelperConfig.
  --> script/deploy.s.sol:13:59:
   |
13 |         HelperConfig.NetworkConfig memory networkConfig = helperConfig.getActiveNetworkConfig();
   |                                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

root@Anon:~/SCL# forge script script/deploy.s.sol   --rpc-url $ETH_SEPOLIA_URL   --private-key $PRIAVTE_KEY   --broadcast
[⠆] Compiling...
[⠰] Compiling 30 files with Solc 0.8.27
[⠔] Solc 0.8.27 finished in 1.99s
Error: Compiler run failed:
Error (9582): Member "getActiveNetworkConfig" not found or not visible after argument-dependent lookup in contract HelperConfig.
  --> script/deploy.s.sol:13:59:
   |
13 |         HelperConfig.NetworkConfig memory networkConfig = helperConfig.getActiveNetworkConfig();
   |                                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

root@Anon:~/SCL# 
```

**Reply by alhenaai** [reply]
Thank you for your question! I’m unable to provide an answer at this moment, but I have forwarded your inquiry to our team. Someone will be with you shortly to assist further. Your patience is appreciated!

**Reply by sanu2002** [reply]
Thanks !

**Comment by JackHamer09** [comment]
Hey, @sanu2002! Looking into your issue now!

**Reply by JackHamer09** [reply]
@sanu2002 Can you try changing `getActiveNetworkConfig` visibility to `public`:
```sol
function getActiveNetworkConfig() public view returns (NetworkConfig memory) {
    return activeNetworkConfig;
}
```

Explanation:
 - External vs. Public:
   - external functions can only be called from outside the contract, e.g., via a transaction or an external contract call.
   - public functions can be called both internally (within the same contract or other contracts) and externally.
- Since DeployRaffle calls the getActiveNetworkConfig function internally, you need it to be public.

Let me know how it goes!

**Reply by sanu2002** [reply]
```
	function getActiveNetworkConfig() public view returns (NetworkConfig memory) {
			return activeNetworkConfig;
	}
```





root@Anon:~/SCL# forge script script/deploy.s.sol   --rpc-url $ETH_SEPOLIA_URL   --private-key $PRIAVTE_KEY   --broadcast
[⠒] Compiling...
[⠒] Compiling 30 files with Solc 0.8.27
[⠢] Solc 0.8.27 finished in 927.54ms
Error: Compiler run failed:
Error (9582): Member "getActiveNetworkConfig" not found or not visible after argument-dependent lookup in contract HelperConfig.
  --> script/deploy.s.sol:13:59:
   |
13 |         HelperConfig.NetworkConfig memory networkConfig = helperConfig.getActiveNetworkConfig();
   |                                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

root@Anon:~/SCL# 






still it's showing me the same i have changed the visibility from external to public .

**Reply by JackHamer09** [reply]
This looks like a solidity related error. I would recommend reading solidity documentation or asking ChatGPT or similar tools with the help, if you haven't yet.
Otherwise providing repository with reproducible example would be helpful, so that I can help you debug this.

**Reply by sanu2002** [reply]
ok i am pushing this file into github  can you please check it out because chatgpt not able resolve this errro and giving me the same ans again and again ..

**Reply by sanu2002** [reply]
> This looks like a solidity related error. I would recommend reading solidity documentation or asking ChatGPT or similar tools with the help, if you haven't yet. Otherwise providing repository with reproducible example would be helpful, so that I can help you debug this.

https://github.com/sanu2002/checkup( Here is the repo link )

**Reply by JackHamer09** [reply]
Sorry, @sanu2002, I just noticed that you're trying to deploy to Ethereum L1 and don't have anything related to ZKsync.
This means it's gotta be something with your code and not tooling issue. Unfortunately we don't provide support for solidity contracts coding. I would recommend to seek for help is Solidity specialized channels.

**Reply by sanu2002** [reply]
ok thanks
